[CI] ClassificationHousePricingIT testFeatureImportanceValues failing

**Build Scans:**
- [elasticsearch-intake #40733 / part4](https://gradle-enterprise.elastic.co/s/d5wkmhqju45xg)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest" --tests "org.elasticsearch.xpack.ml.integration.ClassificationHousePricingIT.testFeatureImportanceValues" -Dtests.seed=3B11DFCA6EBC961C -Dtests.locale=zh-Hans-SG -Dtests.timezone=Etc/Universal -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ml.integration.ClassificationHousePricingIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testFeatureImportanceValues'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: is <true>
     but: was <false>
```

**Issue Reasons:**
- [main] 1 consecutive failures in step part4
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

Assigning `medium-risk` — possible production explainability gap with limited impact.

**Failure:** `ClassificationHousePricingIT.testFeatureImportanceValues` — `AssertionError: Expected: is <true> but: was <false>` on `resultsObject.containsKey("feature_importance")`.

**What the test checks:** After house-pricing classification DFA with `num_top_feature_importance_values: 5`, every source document's dest result must include a non-empty `feature_importance` array. This IT guards #88536 (missing feature importance on this dataset).

**Production behavior:** `ClassificationInferenceResults` omits `feature_importance` when the computed list is empty — so a missing key means per-document importance was empty for at least one doc, not that inference failed.

**Risk assessment (Q1–Q3):**
- Q1: Uncertain production bug — real product path, but history of intermittent empty importance (fixed seed in #145162; see also #124341).
- Q2: Not blocker — predictions still written; no data loss or stability impact.
- Q3: Acceptable to release — supplementary explainability only; 1 consecutive failure; muted on main pending fix.

**Next steps for Team:ML:**
1. Confirm failure rate via CI history (Java 26 / intake part4).
2. Inspect build scan for which doc(s) lack `feature_importance` and whether the trained model differs from the seed-42 baseline.
3. If Java 26–specific non-determinism in ml-cpp, stabilize test or fix upstream; if systematic product gap, ensure importance is always populated when configured.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -389,9 +389,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.recovery.BlockRefreshUponIndexCreationIT
   method: testRefreshesAreBlockedUntilBlockIsRemoved
   issue: https://github.com/elastic/elasticsearch/issues/150492
-- class: org.elasticsearch.xpack.ml.integration.ClassificationHousePricingIT
-  method: testFeatureImportanceValues
-  issue: https://github.com/elastic/elasticsearch/issues/150493
 - class: org.elasticsearch.xpack.esql.analysis.AnalyzerUnmappedGoldenTests
   method: testTSTypeConflictTimeseriesStatsWithCast
   issue: https://github.com/elastic/elasticsearch/issues/150495

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/ClassificationHousePricingIT.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/ClassificationHousePricingIT.java
@@ -1508,14 +1508,6 @@ public class ClassificationHousePricingIT extends MlNativeDataFrameAnalyticsInte
 
     static final String TARGET_FIELD = "CentralAir";
 
-    /**
-     * Fixed seed for boosted-tree training in {@link #testFeatureImportanceValues}. A random seed can rarely
-     * produce empty per-document feature importance; in that case {@code feature_importance} is omitted from
-     * inference results (see {@code ClassificationInferenceResults#addSupportingFieldsToMap}), failing the test.
-     * See <a href="https://github.com/elastic/elasticsearch/issues/124341">#124341</a>.
-     */
-    private static final long FEATURE_IMPORTANCE_RANDOMIZE_SEED = 42L;
-
     private String jobId;
     private String sourceIndex;
     private String destIndex;
@@ -1541,6 +1533,11 @@ public class ClassificationHousePricingIT extends MlNativeDataFrameAnalyticsInte
         String predictionField = TARGET_FIELD + "_prediction";
         initialize("classification_house_pricing_test_feature_importance_values");
         indexData(sourceIndex);
+        // gamma=0 disables the tree-size regularization penalty and the 70% training fraction
+        // keeps much of the imbalanced minority class in the training set; together they prevent
+        // boosted-tree training from occasionally collapsing to a degenerate single-node forest
+        // (constant predictor), which yields empty per-document feature_importance and fails the
+        // assertions below. See https://github.com/elastic/elasticsearch/issues/124341
         DataFrameAnalyticsConfig config = buildAnalytics(
             jobId,
             sourceIndex,
@@ -1548,12 +1545,12 @@ public class ClassificationHousePricingIT extends MlNativeDataFrameAnalyticsInte
             null,
             new Classification(
                 TARGET_FIELD,
-                BoostedTreeParams.builder().setNumTopFeatureImportanceValues(5).build(),
+                BoostedTreeParams.builder().setNumTopFeatureImportanceValues(5).setGamma(0.0).build(),
                 null,
                 null,
                 null,
-                35.0,
-                FEATURE_IMPORTANCE_RANDOMIZE_SEED,
+                70.0,
+                null,
                 null,
                 null
             )

